diff --git a/start_ovs_ovn.sh b/start_ovs_ovn.sh
index 597d8c8..4b2eccc 100644
--- a/start_ovs_ovn.sh
+++ b/start_ovs_ovn.sh
@@ -19,6 +19,9 @@ ovs-ctl restart --no-ovsdb-server  --system-id=random
 ovs-ctl --protocol=udp --dport=6081 enable-protocol
 
 
+# Start northd
+ovn-ctl --db-nb-create-insecure-remote=yes --db-sb-create-insecure-remote=yes start_northd
+
 # Start ovn-controller
 ovn-ctl restart_controller
 ovn-ctl restart_controller_vtep
@@ -30,6 +33,18 @@ ovs-vsctl set open . external-ids:ovn-remote-probe-interval=10000
 ovs-vsctl set open . external-ids:ovn-encap-type=geneve
 
 
+# Add local-ovs-phy port to local switch
+ovn-nbctl ls-add local
+
+# Add local-ovs-phy port to local switch
+ovn-nbctl --may-exist lsp-add local local-ovs-phy
+
+# Configure local-ovs-phy port
+ovn-nbctl lsp-set-addresses local-ovs-phy unknown
+ovn-nbctl lsp-set-type local-ovs-phy localnet
+ovn-nbctl lsp-set-options local-ovs-phy network_name=local-network
+
+
 tail -f /var/log/openvswitch/ovs-vswitchd.log
 
 
