diff --git a/start-ovs-dpdk.sh b/start-ovs-dpdk.sh
index 851961a..44fa64f 100755
--- a/start-ovs-dpdk.sh
+++ b/start-ovs-dpdk.sh
@@ -46,6 +46,12 @@ if [[ `nproc` -gt 12 ]]; then
 fi
 
 # Start ovsdb
+
+ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-socket-mem="1024,0"
+ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-init=true
+ovs-vsctl --no-wait set Open_vSwitch . other_config:pmd-cpu-mask=0x4
+ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-lcore-mask=0x2
+
 /usr/share/openvswitch/scripts/ovs-ctl restart --no-ovsdb-server  --system-id=random
 /usr/share/openvswitch/scripts/ovs-ctl --protocol=udp --dport=6081 enable-protocol
 
